backend/src/app.ts

```diff
@@ -57,6 +57,34 @@ app.post(
   }
 );
 
+/**
+ * User sign up/sign in
+ */
+app.post(
+  '/userLoginSignup',
+  [middleware.authenticateUser, middleware.getDB],
+  async (req: express.Request, res: express.Response) => {
+    // search for existing user
+    console.log(req.user);
+    let user = await getUser(req.db, req.user.oid);
+
+    // create if not found
+    if (user === undefined) {
```

Comment:
A really cool thing about Javascript is that `user ==== undefined` is the exact same thing as saying `!user`. This is more readable:
```
if (!user) {
// create user
}
```